go_metrics/metrics/graphite_time_parser.py

```diff
@@ -0,0 +1,83 @@
+"""
+Time period and interval parameter parsers for Graphite backend.
+"""
+
+from datetime import timedelta
+import re
+
+
+class TimeParserValueError(ValueError):
+    """
+    A value could not be parsed.
+    """
+
+
+INTERVAL_RE = re.compile(r'(?P<count>\d+)(?P<unit>.+)')
+
+UNIT_VALUES = {}
+
+
+def _set_unit_value(value, *names):
+    for name in names:
+        UNIT_VALUES[name] = value
+
+
+_set_unit_value(1, "s", "second", "seconds")
+_set_unit_value(60, "min", "minute", "minutes")
+_set_unit_value(3600, "h", "hour", "hours")
+_set_unit_value(86400, "d", "day", "days")
+_set_unit_value(7 * 86400, "w", "week", "weeks")
+_set_unit_value(30 * 86400, "mon", "month", "months")
+_set_unit_value(365 * 86400, "y", "year", "years")
+
+
+def interval_to_seconds(interval_str):
+    """
+    Parse a time interval specifier of the form "<count><unit>" into the
+    number of seconds contained in the interval.
+
+    NOTE: This is stricter than Graphite's parser, which accepts any string
+          starting with the shortest prefix for a unit.
+    """
```

Comment:
If we are differentiating from graphite's parser, we should change our docs to mention what values we accept.